src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -823,6 +851,15 @@ RuntimeMethodHandleInternal Value
         {
             get;
         }
+
+        internal static IRuntimeMethodInfo Create(IntPtr ptr) => new IRuntimeMethodInfoImp(ptr);
+
+        private sealed class IRuntimeMethodInfoImp : IRuntimeMethodInfo
```

Comment:
I do not think that this implementation is going to work. Some place in reflection assume that types implementing have certain fields at certain offsets - check `RuntimeMethodInfoStub`. This needs to use `RuntimeMethodInfoStub` or something very similar.

(Also, add test that will hit this bug.)

Reply:
Hmmm. I need to look into this more. I'm not entirely sure how this works and where these assumptions are made.